apps/src/templates/CompletionButton.jsx

```diff
@@ -0,0 +1,77 @@
+import msg from '@cdo/locale';
+import React from 'react';
+import {connect} from 'react-redux';
+import ProtectedStatefulDiv from '../templates/ProtectedStatefulDiv';
+
+const styles = {
+  main: {
+    display: 'inline'
+  }
+};
+
+/**
+ * Component for for potentially showing a completion button in applab.
+ * Depending on props, this button (if it exists) will say "Finish", "Submit",
+ * or "Unsubmit"
+ */
+var CompletionButton = React.createClass({
+  propTypes: {
+    isProjectLevel: React.PropTypes.bool.isRequired,
+    isSubmittable: React.PropTypes.bool.isRequired,
+    isSubmitted: React.PropTypes.bool.isRequired
+  },
+
+  render: function () {
+    var id;
+    var contents;
+    var divClass = 'share-cell-none';
```

Comment:
These should all be let/const (I suspect this is code that you're moving instead of new code, but it's green in github, so it's getting reviewed ;-))